### PR TITLE
TemplateLiteralMatch populates its own importPath and importIdentifier

### PR DESCRIPTION
**Pull request by nlfurniss**
This is a PoC of how to do #42 and help unblock template imports in ember-template-lint.

* My approach populates the import path and the import name in a `TemplateLiteralMatch`. The only needed change was to switch `importedNames` from an array to a map so we keep the association between import name and path.

cc: @rwjblue @chriskrycho

### CODE REVIEW

File: src/parse-templates.ts [hunk 1]
```diff
@@ -359,37 +325,30 @@ export function parseTemplates(
...
-      importConfig,
-      $import.moduleName
+    const importIdentifiers = ['default', 'hbs', 'precompileTemplate'];
+    const importModuleNames = [
+      'ember-cli-htmlbars',
+      '@ember/template-compilation',
+      'ember-template-imports',
+      'htmlbars-inline-precompile',
+      'ember-cli-htmlbars-inline-precompile',
+    ];
```
Comment (rwjblue) [inline src/parse-templates.ts:338]: I think we still want these to come from the caller like before (`StaticImportConfig[]`)

File: src/parse-templates.ts [hunk 2]
```diff
@@ -95,21 +68,14 @@ function isEscaped(template: string, _offset: number | undefined) {
  */
 export function parseTemplates(
   template: string,
-  relativePath: string,
-  options?: ParseTemplatesOptions
```
Comment (rwjblue) [inline src/parse-templates.ts:99]: I think we still probably want to accept options like before (basically the same signature as original), but update `TemplateLiteralMatch` to include more metadata:

```ts
export interface TemplateLiteralMatch {
  type: 'template-literal';
  tagName: string;
  importConfig: StaticImportConfig;
  start: RegExpMatchArray;
  end: RegExpMatchArray;
}
```

File: src/parse-templates.ts (hunk 1 above)
Comment (rwjblue) [inline src/parse-templates.ts:338]: FWIW, even if we do it this way, I don't think this logic is correct. For example `import hbs from 'ember-cli-htmlbars'` is not a valid thing to match.

**Comment by nlfurniss** [comment]
@rwjblue I think I misunderstood 42 then. Do we just want to pass the info coming from `ParseTemplatesOptions` and populate `importPath` and `importIdentifier` in `TemplateLiteralMatch`?

**Comment by rwjblue** [comment]
> Do we just want to pass the info coming from ParseTemplatesOptions and populate importPath and importIdentifier in TemplateLiteralMatch?

Yes, I think so...

**Pull request closed without merging**